python/lsst/ts/standardscripts/auxtel/latiss_take_sequence.py

```diff
@@ -0,0 +1,295 @@
+                  Optional name of the program this dataset belongs to.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+
+              do_track_target:
+                description: Track target during sequence.
+                type: boolean
+                default: True
+
+              do_check_ataos_corrections:
+                description: Check that ATAOS corrections are enabled before taking sequence.
+                type: boolean
+                default: True
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        # make a list of tuples from the filter, exptime and grating lists
+        _recurrences = (
+            len(config.exposure_time_sequence)
+            if isinstance(config.exposure_time_sequence, collections.abc.Iterable)
+            else 1
+        )
+
+        self.visit_configs = [
+            (f, e, g)
+            for f, e, g in zip(
+                format_as_list(config.filter_sequence, _recurrences),
+                format_as_list(config.exposure_time_sequence, _recurrences),
+                format_as_list(config.grating_sequence, _recurrences),
+            )
+        ]
+
+        self.reason = config.reason
+
+        if self.reason is None:
+            warnings.warn(
+                "Reason not specified! Consider specifying the 'reason' for "
+                "this observation. it helps reducing the data aftwards. Running "
+                "without specifying a 'reason' will stop working in the future!",
+                DeprecationWarning,
+            )
+
+        self.program = config.program
+
+        if self.program is None:
+            warnings.warn(
+                "Program not specified! Consider specifying the 'program' for "
+                "this observation, it helps reducing the data aftwards. Running "
+                "without specifying a 'program' will stop working in the future!",
+                DeprecationWarning,
+            )
+
+        self.do_track_target = config.do_track_target
+
+        self.do_check_ataos_corrections = config.do_check_ataos_corrections
+
+    def set_metadata(self, metadata):
+        metadata.duration = 0
+        filters, gratings, exptime_total = set(), set(), 0
+        for filt, exptime, grating in self.visit_configs:
+            exptime_total += exptime
+            filters.add(filt)
+            gratings.add(grating)
+            metadata.duration += 3  # time to reconfigure latiss
+        metadata.duration += exptime_total
+        metadata.filter = f"{filters},{gratings}"
+
+    async def take_sequence(self):
+        """Take data while making sure ATCS is tracking if do_track_target."""
+
+        if self.do_track_target:
```

Comment:
I think we should remove this conditional here. Then, before attempting to take data, we should probably check that the telescope is tracking. 

In short:

1. check if we are tracking
2. take sequence and make sure we continue to track.

Reply:
Conditional removed as part of latest commits, is the tasks grouping (L177) enough to do the check before and check while, or are you recommending I add another, separate check?

Reply:
I think this is good for now.. But, I think we will want to make it a bit better in the future, adding a `self.atcs.assert_tracking()`, which currently does not exists.

Can you create a ticket to "Create a method in `BaseTCS` to assert that the telescope is tracking."? Then add a TODO here:

```
# TODO (DM-12345): Call self.atcs.assert_tracking().
```

Reply:
Ticket created, see https://jira.lsstcorp.org/browse/DM-38822. After creating the ticket, I also found https://jira.lsstcorp.org/browse/DM-24488, which I think encompasses this method (and maybe some more?). I linked the two tickets in JIRA, but we may want to deprecate one of them.